pkg/operator/observe_config_test.go

```diff
@@ -0,0 +1,68 @@
+package operator
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+)
+
+func TestObserveClusterConfig(t *testing.T) {
+
+	const (
+		podCIDR     = "10.9.8.7/99"
+		serviceCIDR = "11.6.7.5/88"
+	)
+
+	t.Run("HappyPath", func(t *testing.T) {
+		kubeClient := fake.NewSimpleClientset()
+		cm := &v1.ConfigMap{
+			Data: map[string]string{
+				"install-config": fmt.Sprintf("networking:\n  podCIDR: %s\n  serviceCIDR: %s\n", podCIDR, serviceCIDR),
+			},
+		}
+		cm.Name = "cluster-config-v1"
+		kubeClient.CoreV1().ConfigMaps("kube-system").Create(cm)
+		result, err := observeClusterConfig(kubeClient, rest.Config{}, map[string]interface{}{})
+		if err != nil {
+			t.Error("expected err == nil")
+		}
+		if result == nil {
+			t.Fatal("expected result != nil")
+		}
+		a, ok := result["admissionPluginConfig"]
```

Comment:
just do a straight sequence of map accesses and casts.  If anything fails, we'll chase the panic.